src/device/router_device_channels_worker.erl

```diff
@@ -543,13 +552,55 @@ downlink_decode(MapPayload) when is_map(MapPayload) ->
 downlink_decode(Payload) ->
     {error, {not_binary_or_map, Payload}}.
 
--spec send_to_channel(
+-spec send_join_to_channel(
+    JoinCache :: #join_cache{},
+    Device :: router_device:device(),
+    EventMgrRef :: pid(),
+    Blockchain :: blockchain:blockchain()
+) -> {ok, map()}.
+send_join_to_channel(
+    #join_cache{
+        uuid = UUID,
+        packet_selected = {_Packet0, _PubKeyBin, _Region, PacketTime, _HoldTime} = SelectedPacket,
+        packets = CollectedPackets
+    },
+    Device,
+    EventMgrRef,
+    Blockchain
+) ->
+    FormatHotspot = fun({Packet, PubKeyBin, Region, Time, HoldTime}) ->
+        format_hotspot(PubKeyBin, Packet, Region, Time, HoldTime, Blockchain)
+    end,
+
+    %% No touchy, this is set in STONE
+    Map = #{
+        %% TODO: How do we tell integrations the difference between joins and frames?
```

Comment:
I think we should and add it to normal data also as 'type: packet', maybe?

Reply:
Fixed in commit 5d2e93a